stdlib/public/stubs/LibcShims.cpp

```diff
@@ -18,8 +18,17 @@
 #include <string.h>
 #include "../SwiftShims/LibcShims.h"
 
+#ifdef __ANDROID__
+extern "C" {
+extern size_t dlmalloc_usable_size(void*);
```

Comment:
I would recommend sinking this forward declaration down to its only point of use below.  That will reduce the number of `#if`s and will keep the code focused.